h5py/_hl/group.py

```diff
@@ -258,7 +283,7 @@ def require_dataset(self, name, shape, dtype, exact=False, **kwds):
             except KeyError:
                 dset = self[name]
                 raise TypeError("Incompatible object (%s) already exists" % dset.__class__.__name__)
-            except:
+            except Exception:
                 raise
```

Comment:
Why are we catching and reraising exceptions here, and is switching to capturing only some now safe (as not everything subclasses from `Exception`)? Can we just delete these two lines?

Reply:
Explicitly stating `Exception` is considered a better practice. This will not capture only _SystemExit_, _KeyboardInterrupt_, and _GeneratorExit_.

I am okay with removing this except clause as it seems redundant.

Reply:
Removed it in https://github.com/h5py/h5py/pull/2127/commits/35413a4ba7e447c7a85cbe390e3e87b1273c215c.